src/popup/components/Modals/FormSelectOptions.vue

```diff
@@ -0,0 +1,102 @@
+<template>
+  <Modal
+    class="form-select-options"
+    from-bottom
+    has-close-button
+    semi-dense
+    @close="reject()"
+  >
+    <template #header>
+      <div class="header">
+        <span class="text-heading-3 title">{{ title }}</span>
+        <InputSearch
+          v-if="options.length > 3"
+          v-model="searchTerm"
+          class="search-field"
+          :placeholder="$t('search')"
+        />
+      </div>
+    </template>
+
+    <div>
+      <ListItemWrapper
+        v-for="(option, index) in optionsFiltered"
+        :key="index"
+        :selected="(option.value === value)"
+        class="option"
+        @click.prevent="resolve(option.value)"
+      >
+        {{ option.text }}
+      </ListItemWrapper>
+    </div>
+  </Modal>
+</template>
+
+<script lang="ts">
+import {
+  computed,
+  defineComponent,
+  PropType,
+  ref,
+} from '@vue/composition-api';
+import type { FormSelectOption } from '../form/FormSelect.vue';
+import InputSearch from '../InputSearch.vue';
+import Modal from '../Modal.vue';
+import ListItemWrapper from '../ListItemWrapper.vue';
+
+export default defineComponent({
+  components: {
+    Modal,
+    ListItemWrapper,
+    InputSearch,
+  },
+  props: {
+    // eslint-disable-next-line no-unused-vars
+    resolve: { type: Function as PropType<(o: any) => any>, required: true },
+    reject: { type: Function, required: true },
+    title: { type: String, default: null },
+    value: { type: [String, Number], default: null },
+    options: { type: Array as PropType<FormSelectOption[]>, default: () => [] },
+  },
+  setup(props) {
+    const searchTerm = ref('');
+    const optionsFiltered = computed((): FormSelectOption[] => {
+      if (searchTerm.value) {
+        const searchTermLower = searchTerm.value.toLowerCase();
+        return props.options.filter(({ text, value }) => (
+          text.toLowerCase().includes(searchTermLower)
+          || value.toString().includes(searchTermLower)
```

Comment:
We have a function called `compareCaseInsensitive` - maybe it could be extended for not exact comparison?

Reply:
Mentioned function is lowering both strings each time its fired. Here I'm storing the parsed search term for the performance reasons.

Reply:
So actually I see that Konrad's code needs to be refactored :)